.github/workflows/build-image.yml

```diff
@@ -10,32 +10,81 @@ on:
     paths:
       - .github/workflows/build-image.yml
       - Dockerfile
+
 permissions:
   contents: read
+  packages: write
+
+env:
+  IMAGE_NAME: ${{ github.repository }}
 
 jobs:
   build-image:
     runs-on: ubuntu-latest
     steps:
+      - name: Extract release name
+        env:
+          INPUT_IMAGE_NAME: ${{ secrets.IMAGE_NAME }}
+        shell: bash
+        run: |
+          if [ -z "$INPUT_IMAGE_NAME" ]; then
+              if [[ "$IMAGE_NAME" == "mastodon/mastodon" ]]; then
+                  image_name=tootsuite/mastodon
+              else
+                  image_name=$IMAGE_NAME
+              fi
+          else
+              image_name=$INPUT_IMAGE_NAME
+          fi
+          echo IMAGE_NAME=$image_name >> $GITHUB_ENV
+        id: extract_image_name
+
+      - name: Configure Docker
```

Comment:
Why all this?

Reply:
If you mean the "Configure Docker" part? It's because I want in my repo to publish to github's image repository, not dockerhub, so there needs to be a way to choose a different registry.

If the image is published to github's registry, authentication is pretty simple because the actions token will have push permissions already thanks to the grant on line 16. Otherwise, we need the whole set of credentials provided in secrets.